simple_graph.py

```diff
@@ -37,24 +49,109 @@ def del_edge(self, node1, node2):
             raise KeyError("{} does not exist".format(node1))
         try:
             temp.remove(node2)
-        except KeyError:
-            raise KeyError("{} does not exist".format(node2))
+        except ValueError:
+            raise ValueError("{} does not exist".format(node2))
 
     def has_node(self, node):
+        """Return True if node is in graph."""
         return node in self.graph
 
     def neighbors(self, node):
+        """Return a list of nodes connected to the supplied node. Throw a
+         KeyError if the supplied node wasn't in the graph."""
         try:
             return self.graph[node]
         except KeyError:
             raise KeyError("{} doesn't exist".format(node))
 
     def adjacent(self, node1, node2):
+        """Return true if an edge points from the first node to the second
+         node. Throw a KeyError if the first node wasn't in the graph."""
         try:
             return node2 in self.graph[node1]
         except KeyError:
             raise KeyError("{} doesn't exist".format(node1))
 
+    def _depth_first_helper(self, node, visited):
```

Comment:
There's a more efficient method than recursion to do this. Think about a simple data structure that could do the same